Classes/DDLog.m

```diff
@@ -1107,15 +1107,16 @@ - (instancetype)initWithMessage:(NSString *)message
                         options:(DDLogMessageOptions)options
                       timestamp:(NSDate *)timestamp {
     if ((self = [super init])) {
-        _message      = [message copy];
+        BOOL copyMessage = (options & DDLogMessageDontCopyMessage) == 0;
+        _message      = copyMessage ? [message copy] : message;
```

Comment:
If we want performance then we probably don't need an extra variable and can just do:

`_message      = (options & DDLogMessageDontCopyMessage) ? message : [message copy];`